src/shogun/lib/DynamicArray.h

```diff
@@ -622,20 +731,13 @@ template <class T> class CDynamicArray :public CSGObject
 		{
 			set_generic<T>();
 
-			m_parameters->add_vector(&m_array.array,
-					&m_array.current_num_elements, "array",
-					"Memory for dynamic array.");
-			SG_ADD(&m_array.resize_granularity,
-							  "resize_granularity",
-							  "shrink/grow step size.", MS_NOT_AVAILABLE);
-			SG_ADD(&m_array.use_sg_mallocs,
-							  "use_sg_malloc",
-							  "whether SG_MALLOC or malloc should be used",
-							  MS_NOT_AVAILABLE);
-			SG_ADD(&m_array.free_array,
-							  "free_array",
-							  "whether array must be freed",
-							  MS_NOT_AVAILABLE);
+			SG_ADD(
```

Comment:
by removing
```
m_parameters->add_vector(&m_array.array, &m_array.current_num_elements, "array", "Memory for dynamic array.");
```
basically it means that the serialization of this object will not be correct... as the deque is not going to be serialized.
it would be easy to add a vector element to `m_parameters` if we would use `std::vector` as that is a continuous memory block... unfortunately it's not the case of deque. but we still need a way to be able to serialize deque's content :S

@lisitsyn btw how is this gonna be handled with tags? `m_parameters->add_vector`